src/shared/reducer.ts

```diff
@@ -33,3 +34,102 @@ export const reducer = combineReducers({
     activeMusic: ephemeralMusicReducer,
   }),
 });
+
+const onInitiativeChangedCharactersReducer = createReducer(
+  initialSyncedState,
+  (builder) => {
+    builder.addCase(initiativeTrackerSetCurrentEntry, (state, action) => {
+      if (!state.initiativeTracker.currentEntryId) return;
+      const newInitiativeEntry =
+        state.initiativeTracker.entries.entities[
+          state.initiativeTracker.currentEntryId
+        ];
+      if (state.initiativeTracker.currentEntryId !== action.payload) {
+        throw new Error(
+          "Current initiative entry does not match action payload.\
+        This should have been updated by the single slice reducers already."
+        );
+      }
+      if (!newInitiativeEntry || newInitiativeEntry.type === "lairAction")
+        return;
+      newInitiativeEntry.characterIds.map((id) => {
+        const characterEntry = state.characters.entities[id];
+        if (characterEntry?.currentlyConcentratingOn) {
+          characterEntry.currentlyConcentratingOn.roundsLeft--;
+        }
+      });
+    });
+  }
+);
+
+const beforeInitiativeChangedCharactersReducer = createReducer(
+  initialSyncedState,
+  (builder) => {
+    builder.addCase(initiativeTrackerSetCurrentEntry, (state, action) => {
+      if (!state.initiativeTracker.currentEntryId) return;
+      const currentInitiativeEntry =
+        state.initiativeTracker.entries.entities[
+          state.initiativeTracker.currentEntryId
+        ];
+      if (state.initiativeTracker.currentEntryId === action.payload) {
+        //TODO do we need this? It would fail if we only have one entry I assume
+        throw new Error(
+          "Current initiative entry matches action payload.\
+          This should run before the initiative entry is updated."
+        );
```

Comment:
Oh yeah, I think we should just return the state unchanged here instead of throwing an error.